plugins/inventory/opennebula.py

```diff
@@ -199,6 +199,12 @@ def _retrieve_servers(self, label_filter=None):
                     continue
 
             server['name'] = vm.NAME
+            server['id'] = vm.ID
+            if hasattr(vm.HISTORY_RECORDS, 'HISTORY'):
+                if vm.HISTORY_RECORDS.HISTORY:
+                    if isinstance(vm.HISTORY_RECORDS.HISTORY, list) and len(vm.HISTORY_RECORDS.HISTORY) > 0:
+                        if hasattr(vm.HISTORY_RECORDS.HISTORY[-1], 'HOSTNAME'):
+                            server['host'] = vm.HISTORY_RECORDS.HISTORY[-1].HOSTNAME
```

Comment:
You can leave the first `if` away:
```suggestion
                if isinstance(vm.HISTORY_RECORDS.HISTORY, list) and len(vm.HISTORY_RECORDS.HISTORY) > 0:
                    if hasattr(vm.HISTORY_RECORDS.HISTORY[-1], 'HOSTNAME'):
                        server['host'] = vm.HISTORY_RECORDS.HISTORY[-1].HOSTNAME
```
Also it's better to use `Sequence` instead of `list`, since maybe the library might for some reason use a tuple instead of a list:
```suggestion
                if isinstance(vm.HISTORY_RECORDS.HISTORY, Sequence) and len(vm.HISTORY_RECORDS.HISTORY) > 0:
                    if hasattr(vm.HISTORY_RECORDS.HISTORY[-1], 'HOSTNAME'):
                        server['host'] = vm.HISTORY_RECORDS.HISTORY[-1].HOSTNAME
```
For that to work, you need to add `from ansible.module_utils.common._collections_compat import Sequence` next to the other imports (around line 99/100).